### PR TITLE
Expr supports __array__

### PR DESCRIPTION
**Pull request by mrocklin**
Support standard protocols like `__array__`, `__float__`, etc. to improve standard interoperation with normal Python code.

```Python
In [1]: from blaze import *
x
In [2]: x = np.arange(6)

In [3]: d = Data(x)

In [4]: np.array(d * 10)
Out[4]: array([ 0, 10, 20, 30, 40, 50])

In [5]: int(d.sum())
Out[5]: 15

In [6]: x[:3] = d[3:] * 10

In [7]: x
Out[7]: array([30, 40, 50,  3,  4,  5])
```

Should have done this long ago.

### CODE REVIEW

File: blaze/compute/tests/test_pandas_compute.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def df_all(a_df, b_df):
     This method doesn't check the index which can be manipulated during
     operations.
     """
-    assert all(a_df.columns == b_df.columns)
+    assert np.all(a_df.columns == b_df.columns)
```
Comment (cpcloud) [inline blaze/compute/tests/test_pandas_compute.py]: Any reason we shouldn't gut this function and just use `tm.assert_frame_equal(a_df, b_df)`? The use of this later in this file looks redundant.

File: blaze/expr/tests/test_arithmetic.py [hunk 2]
```diff
@@ -64,8 +64,8 @@ def test_dir():
     i = symbol('i', '10 * int')
     d = symbol('d', '10 * datetime')
 
-    assert i + 1
-    with pytest.raises(Exception):
+    i + 1  # this works
```
Comment (cpcloud) [inline blaze/expr/tests/test_arithmetic.py]: i like having the assert her

File: blaze/expr/tests/test_collections.py [hunk 3]
```diff
@@ -23,7 +24,8 @@ def test_transform():
     expr = transform(t, z=t.x + t.y)
     assert expr.fields == ['x', 'y', 'z']
 
-    assert (t.x + t.y) in expr._subterms()
+    assert builtins.any((t.x + t.y).isidentical(node)
```
Comment (cpcloud) [inline blaze/expr/tests/test_collections.py]: this is one reason i don't like star imports, just a nit though

File: blaze/interactive.py [hunk 4]
```diff
@@ -281,9 +281,24 @@ def into(a, b, **kwargs):
...
+
+def intonumpy(data, dtype=None, **kwargs):
+    # TODO: Don't ignore other kwargs like copy
+    result = into(np.ndarray, data)
+    if dtype:
+        result = result.astype(dtype)
+    return result
+
+Expr.__array__ = intonumpy
+Expr.__int__ = lambda x: int(compute(x))
```
Comment (cpcloud) [inline blaze/interactive.py]: i think these could also be written as `into(int)`, etc. not a big deal, just a comment that can probably be ignored

File: blaze/expr/tests/test_table.py [hunk 5]
```diff
@@ -201,7 +189,7 @@ def test_path_issue():
     t2 = transform(t, sizes=t.result.map(lambda x: (x - MIN)*10/(MAX - MIN),
                                          schema='float64', name='size'))
 
-    assert t2.sizes in t2.children
+    assert builtins.any(t2.sizes.isidentical(node) for node in t2.children)
```
Comment (cpcloud) [inline blaze/expr/tests/test_table.py]: should we just override `__contains__` so that `in` works like this?

Comment (cpcloud) [inline blaze/expr/tests/test_table.py]: hm nevermind, we often want to check a specific list of things, not the actual expression

File: blaze/expr/tests/test_arithmetic.py (hunk 2 above)
Comment (mrocklin) [inline blaze/expr/tests/test_arithmetic.py]: What should we assert?

File: blaze/compute/tests/test_pandas_compute.py (hunk 1 above)
Comment (mrocklin) [inline blaze/compute/tests/test_pandas_compute.py]: Gutted.

File: blaze/interactive.py (hunk 4 above)
Comment (mrocklin) [inline blaze/interactive.py]: We don't have `Expr` in the `convert` network.  Hence the explicit call to `compute`.

File: blaze/expr/tests/test_arithmetic.py (hunk 2 above)
Comment (cpcloud) [inline blaze/expr/tests/test_arithmetic.py]: Nothing in particular, what was wrong with what was there? Bare code that isn't assigned to anything or used anywhere increases the likelihood that I will delete it and that could potentially hide regressions. Likely a personal shortcoming of mine, but I think it's easy just to keep the assert

Comment (mrocklin) [inline blaze/expr/tests/test_arithmetic.py]: `assert` checks the non-zero-ness of the expressions which now triggers a call to `compute`

Comment (cpcloud) [inline blaze/expr/tests/test_arithmetic.py]: How about is not None?

Comment (mrocklin) [inline blaze/expr/tests/test_arithmetic.py]: I've added a type check

File: blaze/expr/tests/test_collections.py (hunk 3 above)
Comment (mrocklin) [inline blaze/expr/tests/test_collections.py]: Ironically this would have worked fine if we had star-imported from `blaze` rather than `blaze.expr`.

**Comment by mrocklin** [comment]
ok to merge?

**Comment by cpcloud** [comment]
fine by me

**Pull request merged**